x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/ManyShardsIT.java

```diff
@@ -297,4 +304,53 @@ public void testCancelUnnecessaryRequests() {
             coordinatorNodeTransport.clearAllRules();
         }
     }
+
+    public void testMultiTypes() {
+        String q0 = """
+            FROM test-* METADATA _index
```

Comment:
Out of curiosity, why do we need several indices to be able to reproduce?

Shouldn't this be reproducible with just one index that is sharded?

The problem arises in `buildBlocks` from `ValuesFromManyReader` right? How is that code dependant on having several indices.

```java
private void buildBlocks() {
    for (int f = 0; f < target.length; f++) {
        // Here entrySet might not be ordered
        for (var entry : buildersAndLoaders.get(f).entrySet()) {
```